### PR TITLE
Fix flakiness in TestRedirectWithConcurrentActivities

### PR DESCRIPTION
**Pull request by ShahabT**
## What changed?

- Prevent setting MS sticky queue if it belongs to an obsolete build ID. (In case WF has moved to a new build ID since the last WFT started).
- Improve sticky queue check in `processBuildIdRedirectInfo` at WFT start time to compare the sticky TQ name and not only if the sticky queue is set in the MS.

## Why?

- Setting obsolete sticky queue can cause workflow tasks to go back to the previous build ID.
- The old code did not catch edge cases when a sticky queue of an older build ID wants to start a WFT when WF is already moved to a sticky queue on a newer build id.

## How did you test it?

Added unit test.

## Potential risks

For this to work properly, I'm assuming all the SDK versions that support versioning will set the task queue kind in poll request.

## Documentation

none.

## Is hotfix candidate?

No.

### CODE REVIEW

File: service/history/api/recordworkflowtaskstarted/api.go [hunk 1]
```diff
@@ -139,6 +139,13 @@ func Invoke(
 			currentTaskQueue := mutableState.CurrentTaskQueue()
 			if currentTaskQueue.Kind == enumspb.TASK_QUEUE_KIND_STICKY &&
 				currentTaskQueue.GetName() != req.PollRequest.TaskQueue.GetName() {
+				// For versioned workflows we additionally check for the poller queue to not be a sticky queue itself.
+				// Although it's ideal to check this for unversioned workflows as well, we can't rely on older clients
+				// setting the poller TQ kind.
+				if mutableState.GetAssignedBuildId() != "" && req.PollRequest.TaskQueue.Kind == enumspb.TASK_QUEUE_KIND_STICKY {
+					return nil, serviceerrors.NewObsoleteDispatchBuildId("wrong sticky queue")
```
Comment (bergundy) [inline service/history/api/recordworkflowtaskstarted/api.go:146]: Do we also need to clear the sticky queue in this situation?

**Review by bergundy — APPROVED** [review]
